### PR TITLE
Add DNS service discovery

### PR DESCRIPTION
**Pull request by mrgnr**
Add DNS service discovery using [dnsdock](https://github.com/tonistiigi/dnsdock). When `docker-compose up` is run, the dnsdock container automatically starts and allows services to discover each other on the .local.edx.org domain. Dnsdock requires that the DNS listening port is bound to the docker0 interface, so the docker daemon must be started with options `--bip=172.17.42.1/24` and `--dns=172.17.42.1` to use dnsdock.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @emrob! I've created [OSPR-871](https://openedx.atlassian.net/browse/OSPR-871) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).
We can't start reviewing your pull request until you've   submitted a [signed contributor agreement](http://open.edx.org/sites/default/files/wysiwyg/individual-contributor-agreement.pdf)   or indicated your institutional affiliation   and   added yourself to the [AUTHORS](https://github.com/edx/configuration/blob/master/AUTHORS) file . Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst) file for more information.

File: docker/compose/docker-compose.yml [hunk 1]
```diff
@@ -1,40 +1,67 @@
...
+  # The docker daemon must be run with the following options:
+  #  --bip=172.17.42.1/24
+  #  --dns=172.17.42.1
+  # See https://github.com/tonistiigi/dnsdock#user-content-setup
+  image: tonistiigi/dnsdock:v1.10.0
+  volumes:
+    - /var/run/docker.sock:/var/run/docker.sock
+  ports:
+    - 172.17.42.1:53:53/udp
+  command: -domain="local.edx.org"
```
Comment (feanil) [inline docker/compose/docker-compose.yml:11]: Is this right, shouldn't there be a command before the parameter?

Comment (mrgnr) [inline docker/compose/docker-compose.yml:11]: It looks a little odd, but that should be correct. Docker images have both an Entrypoint and a Cmd. The Entrypoint is the executable that runs when a container starts and the Cmd is essentially passed to the Entrypoint as an argument (see [here](https://stackoverflow.com/questions/21553353/what-is-the-difference-between-cmd-and-entrypoint-in-a-dockerfile)). Dnsdock's Entrypoint is `/go/bin/dnsdock`, so when the container starts, we're running `/go/bin/dnsdock -domain="local.edx.org"`.

**Comment by sarina** [comment]
@emrob could you rebase your branch?

@feanil FYI we have Morgan's contributor info, this is ready for review.

**Comment by feanil** [comment]
@emrob looks like this needs a rebase, can you do that?

@e0d any objections to merging this into the docker-hack branch?

**Comment by mrgnr** [comment]
@feanil @e0d I rebased and fixed some of the links to make them consistent with how dnsdock [does matching](https://github.com/tonistiigi/dnsdock#user-content-usage). Things like `lms-mysql.local.edx.org` won't match `mysql.local.edx.org`, so I replaced the dashes with dots. Is there a specific reason that we're prepending the DNS name with the service's name? It might make as much sense to leave the service off and just say `mysql.local.edx.org`.

I also fixed some indentation and added links from the CMS and LMS to the forums.

**Comment by e0d** [comment]
@emrob I think we decided this could be closed now in favor of using docker networking

**Comment by mrgnr** [comment]
@e0d Agreed

**Pull request closed without merging**